### PR TITLE
Fix the bug for "bytes remove --end" .

### LINKED ISSUE #11426: `bytes remove --end` causes panic when specified bytes not present
### Describe the bug

When using the `--end` option of bytes remove, nushell panics if the provided bytes don't exist. This doesn't seem to affect `bytes remove` w/o flag or `bytes remove --all`.

### How to reproduce

1. Construct `binary` value
2. Run `bytes remove --end` with some value that does not exist at the end

Example:
`0x[FF] | bytes remove --end 0x[00]`

### Expected behavior

Pass through the binary value unchanged without panicing

### Screenshots

_No response_

### Configuration

| key                | value                                                                     |
| ------------------ | ------------------------------------------------------------------------- |
| version            | 0.86.0                                                                    |
| branch             |                                                                           |
| commit_hash        | 5d8763ed1d14048ac9fc09eb21be36e6e622f0d7                                  |
| build_os           | linux-x86_64                                                              |
| build_target       | x86_64-unknown-linux-gnu                                                  |
| rust_version       | rustc 1.71.1 (eb26296b5 2023-08-03)                                       |
| rust_channel       | 1.71.1-x86_64-unknown-linux-gnu                                           |
| cargo_version      | cargo 1.71.1 (7f1d04c00 2023-07-29)                                       |
| build_time         | 2023-10-17 17:14:05 +00:00                                                |
| build_rust_channel | release                                                                   |
| allocator          | mimalloc                                                                  |
| features           | dataframe, default, extra, sqlite, static-link-openssl, trash, which, zip |
| installed_plugins  | from eml, from ics, from ini, from vcf                                    |

### Additional context

<details>
<summary>Backtrace</summary>
<pre>
$ 0x[FF] | bytes remove --end 0x[00]
thread 'main' panicked at /home/rose/.cargo/registry/src/index.crates.io-6f17d22bba15001f/nu-command-0.88.1/src/bytes/remove.rs:162:31:
range end index 18446744073709551615 out of range for slice of length 1
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::slice::index::slice_end_index_len_fail
   3: nu_command::bytes::remove::remove
   4: nu_protocol::pipeline_data::PipelineData::map
   5: nu_cmd_base::input_handler::operate
   6: <nu_command::bytes::remove::BytesRemove as nu_protocol::engine::command::Command>::run
   7: nu_engine::eval::eval_call
   8: nu_engine::eval::eval_expression_with_input
   9: nu_engine::eval::eval_element_with_input
  10: nu_engine::eval::eval_block
  11: nu_cli::util::eval_source
  12: nu_cli::repl::evaluate_repl
  13: nu::run::run_repl
  14: nu::main
</pre>
</details>

### PR DESCRIPTION
**Pull request by SebastianIonel**
This PR should close #11426 .

# Description
> ### Describe the bug
> When using the `--end` option of bytes remove, nushell panics if the provided bytes don't exist. This doesn't seem to affect `bytes remove` w/o flag or `bytes remove --all`.

# User-Facing Changes
Nushell doesn`t panic anymore.

# Tests + Formatting
Behavior before fixing the bug:
nu-before changes
Behavior after fixing the bug:
nu- after changes

### CODE REVIEW

File: crates/nu-command/src/bytes/remove.rs
Comment (WindSoilder) [inline crates/nu-command/src/bytes/remove.rs:1]: Would you please add a  test for it?

**Review by WindSoilder — CHANGES_REQUESTED** [review]
Hi, thanks for your contribution!  It'd be good to add a test for such behavior.

I think you can add it to examples

File: crates/nu-command/src/bytes/remove.rs
Comment (SebastianIonel) [reply crates/nu-command/src/bytes/remove.rs:1]: I am sorry, but for the moment, I don't really know how to add a test.

Comment (amtoine) [reply crates/nu-command/src/bytes/remove.rs:1]: @SebastianIonel 
you should be able to copy-paste one of the `Example` from https://github.com/nushell/nushell/blob/95220520639910c77d79c31bece37c468d54aeec/crates/nu-command/src/bytes/remove.rs#L85-L120

**Comment by fdncred** [comment]
You could also just add the test @WindSoilder added before closing his PR that does the same thing. https://github.com/nushell/nushell/pull/11427/files

**Review by WindSoilder — APPROVED** [review]
Thanks!  Looks good to me

**Pull request merged**